### PR TITLE
CDAP-15644: Added BigQuery integration tests.

### PR DESCRIPTION
**Pull request by AntonGensitskiy**
JIRA: https://issues.cask.co/browse/CDAP-15644

In scope of this PR:
* Was added base BigQuery integration tests.

### CODE REVIEW

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+  private static void deleteDatasets() {
+    DatasetId datasetId = dataset.getDatasetId();
+    LOG.info("Deleting dataset {}", bigQueryDataset);
+    boolean deleted = bq.delete(datasetId, BigQuery.DatasetDeleteOption.deleteContents());
+    if (deleted) {
+      LOG.info("Deleted dataset {}", bigQueryDataset);
+    }
+  }
+
+  private boolean isExisting(String tableId) {
```
Comment (aonischuk) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java]: Can we call this `private boolean exists(String tableId)` seems like more grammatically correct. This how the method is called in File API for checking if file exists.

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+    LOG.info("Creating dataset {}", bigQueryDataset);
+    dataset = bq.create(datasetInfo);
+    LOG.info("Created dataset {}", bigQueryDataset);
+  }
+
+  private com.google.cloud.bigquery.Schema getTableSchema(TableId tableId) {
+    return bq.getTable(tableId).getDefinition().getSchema();
+  }
+
+  private static List<FieldValue> getResultTableDate(TableId tableId) {
```
Comment (aonischuk) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java]: getResultTableData

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+
+    com.google.cloud.bigquery.Schema resultSchema = getTableSchema(TableId.of(bigQueryDataset, destinationTableName));
+    Assert.assertEquals(expectedSchema, resultSchema);
+
+    List<FieldValue> result = getResultTableDate(TableId.of(bigQueryDataset, destinationTableName));
+    Assert.assertEquals(expectedResult.size(), result.size());
+
+    for (int i = 0; i < result.size(); i++) {
+      Assert.assertEquals(expectedResult.get(i).getAttribute(), result.get(i).getAttribute());
+      Assert.assertEquals(expectedResult.get(i).getValue(), result.get(i).getValue());
```
Comment (aonischuk) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java]: Can we throw a bit more informative exception here. Like which attribute has different values.

Also I see you check these in loop is that guaranteed that the FieldValues are in the same order?

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
```
Comment (aonischuk) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java]: the same

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java [hunk 5]
```diff
@@ -0,0 +1,551 @@
...
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class GoogleBigQueryTest extends DataprocETLTestBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GoogleBigQueryTest.class);
+  private static final String BIG_QUERY_PLUGIN_NAME = "BigQueryTable";
+  private static final String SOURCE_TABLE_NAME_TEMPLATE = "test_source_table_";
+  private static final String SINK_TABLE_NAME_TEMPLATE = "test_sink_table_";
+  private static final String SOURCE_PATH = "src" + System.getProperty("file.separator") + "test"
```
Comment (aonischuk) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java]: Can we use `Thread.currentThread().getContextClassLoader().getResource(path);` instead?
Usually this is how resource files are read.

File: pom.xml [hunk 6]
```diff
@@ -66,6 +66,11 @@
       <artifactId>google-cloud-datastore</artifactId>
       <version>1.61.0</version>
     </dependency>
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-bigquery</artifactId>
+      <version>1.36.0</version>
```
Comment (aonischuk) [inline pom.xml:77]: Any reason for using older version? Instead of 1.83.0

**Review by aonischuk — CHANGES_REQUESTED** [review]

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java [hunk 7]
```diff
@@ -0,0 +1,551 @@
...
+      }
+    }, 5, TimeUnit.MINUTES, 3, TimeUnit.SECONDS);
+  }
+
+  @Override
+  protected void innerTearDown() throws Exception {
+  }
+
+  @Test
+  public void test_01_ReadDataAndStoreInNewTable() throws Exception {
```
Comment (aielchiieva) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java]: Please remove numbers from units test names: `test_01_ReadDataAndStoreInNewTable` -> `testReadDataAndStoreInNewTable`

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java [hunk 8]
```diff
@@ -0,0 +1,551 @@
...
+    final int testId = 1;
+
+    String sourceTableName = SOURCE_TABLE_NAME_TEMPLATE + testId;
+    String destinationTableName = SINK_TABLE_NAME_TEMPLATE + testId;
+
+    Table sourceTable = createTestTable(bigQueryDataset, sourceTableName, getSimpleFieldsSchema());
+    insertData(bigQueryDataset, sourceTableName, Paths.get(SOURCE_PATH + "simple_source.json"));
```
Comment (aielchiieva) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java]: `Paths.get(SOURCE_PATH, "simple_source.json")`

File: integration-test-remote/src/test/resources/full_source.json [hunk 9]
```diff
@@ -0,0 +1 @@
+{"string_value": "string_1","int_value": 1,"float_value": 0.1,"boolean_value": true,"numeric_value": 123.456,"timestamp_value": "2014-08-01 12:41:35.220000+00:00","date_value": "2014-08-01","time_value": "01:41:35.220000","datetime_value": "2014-08-01 01:41:35.220000","string_array": ["a_string_1", "a_string_2"]}
```
Comment (aielchiieva) [inline integration-test-remote/src/test/resources/full_source.json]: When writing tests for Datastore during code review I was asked to remove JSON files and store data in class, please check Datastore tests.

**Pull request closed without merging**